include/envoy/config/subscription.h

```diff
@@ -179,11 +179,8 @@ class Subscription {
    * Start a configuration subscription asynchronously. This should be called once and will continue
    * to fetch throughout the lifetime of the Subscription object.
    * @param resources set of resource names to fetch.
-   * @param use_namespace_matching if the subscription is for a collection of resources. In such a
-   * case a namespace watch will be created.
    */
-  virtual void start(const std::set<std::string>& resource_names,
-                     const bool use_namespace_matching = false) PURE;
+  virtual void start(const std::set<std::string>& resource_names) PURE;
```

Comment:
Just to get a better understanding of this, the change sets the single namespace matching to be per-subscription, instead of per-resource. I see that VHDS uses the same namespace matching to be per-subscription.
Would there be cases that need different namespaces for different resource names using the same subscription (maybe using a relay)? If so I think this change requires multiple streams, each with different namespace, between a client and server.

Reply:
Good point. For LDS, CDS, SRDS, VHDS, I think this is always set once per subscription. The more I think about it though, the whole namespace matching thing really is only needed for VHDS, as we always know at subscription and resource delivery time whether we are dealing with an `xdstp://` resource via prefix. So, I think I'm going to make a small change to keep it for VHDS only, and use prefix internally within `WatchMap` etc. This basically simplifies things for when we do RDS, EDS, as you don't know ahead of time if you are doing this kind of matching.